test/upgrade/2.0.0-2.1.0.js

```diff
@@ -0,0 +1,587 @@
+        // Convert currency unit from ether to wei
+        value: ethers.utils.parseEther("1"),
+      };
+      await deployer.sendTransaction(tx);
+
+      // create entities
+      switch (creationOrder[i]) {
+        case entity.DR: {
+          const disputeResolver = mockDisputeResolver(wallet.address, wallet.address, wallet.address, wallet.address);
+          const disputeResolverFees = [
+            new DisputeResolverFee(ethers.constants.AddressZero, "Native", "0"),
+            new DisputeResolverFee(mockToken.address, "MockToken", 0),
+          ];
+          const sellerAllowList = [];
+          await accountHandler
+            .connect(connectedWallet)
+            .createDisputeResolver(disputeResolver, disputeResolverFees, sellerAllowList);
+          DRs.push({ wallet: connectedWallet, disputeResolver, disputeResolverFees, sellerAllowList });
+
+          //ADMIN role activates Dispute Resolver
+          await accountHandler.connect(deployer).activateDisputeResolver(disputeResolver.id);
+          break;
+        }
+        case entity.SELLER: {
+          const seller = mockSeller(wallet.address, wallet.address, wallet.address, wallet.address);
+          let authToken;
+          // randomly decide if auth token is used or not
+          if (Math.random() > 0.5) {
+            // no auth token
+            authToken = mockAuthToken();
+          } else {
+            // use auth token
+            seller.admin = ethers.constants.AddressZero;
+            await mockAuthERC721Contract.connect(connectedWallet).mint(101 * i, 1);
+            authToken = new AuthToken(`${101 * i}`, AuthTokenType.Lens);
+          }
+          // set unique new voucherInitValues
+          const voucherInitValues = new VoucherInitValues(`http://seller${i}.com/uri`, i * 10);
+          await accountHandler.connect(connectedWallet).createSeller(seller, authToken, voucherInitValues);
+          sellers.push({ wallet: connectedWallet, seller, authToken, voucherInitValues });
+
+          // mint mock token to sellers just in case they need them
+          await mockToken.mint(connectedWallet.address, "10000000000");
+          await mockToken.connect(connectedWallet).approve(protocolDiamondAddress, "10000000000");
+          break;
+        }
+        case entity.AGENT: {
+          const agent = mockAgent(wallet.address);
+          await accountHandler.connect(connectedWallet).createAgent(agent);
+          agents.push({ wallet: connectedWallet, agent });
+          break;
+        }
+        case entity.BUYER: {
+          // no need to explicitly create buyer, since it's done automatically during commitToOffer
+          const buyer = mockBuyer(wallet.address);
+          buyers.push({ wallet: connectedWallet, buyer });
+          break;
+        }
+      }
+    }
+
+    // create offers - first seller has 5 offers, second 4, third 3 etc
+    let offerId = 0;
+    for (let i = 0; i < sellers.length; i++) {
+      for (let j = i; j >= 0; j--) {
```

Comment:
This doesn't look quite right to me. The first time both loops are executed, `j = i = 0`; the` j `loop will continue as long as` j >= 0`, so it will go into the loop the first time. But` j `gets decremented after the first loop.

Reply:
Combinations of i and j that result in code being executed
| i | j |
| - | - |
| 0 | 0 |
| 1 | 1 |
| 1 | 0 |
| 2 | 2 |
| 2 | 1 |
| 2 | 0 |
| 3 | 3 |
| 3 | 2 |
| 3 | 1 |
| 3 | 0 |
| 4 | 4 |
| 4 | 3 |
| 4 | 2 |
| 4 | 1 |
| 4 | 0 |

And since `j` determines the seller, it means that j=0 is used 5 times, j=1 used 4 times etc. So it's in line with the comment above `// create offers - first seller has 5 offers, second 4, third 3 etc` and it works as intended.